0miner

```diff
@@ -255,7 +255,7 @@ elif [[ $ALGO == X16R || $ALGO == SIB || $ALGO == LYRA2Z || $ALGO == SKUNK || $A
 then
   UCCALGO="-a ${ALGO,,}"
   HCD="${NVOC}"/miners/"${!xminer}"/"${!xversion}"/ccminer
-  echo "$LAUNCH $HCD $UCCALGO -o ${!xproto}://"${!xpool}":"${!xport}" -u "${!xaddr}""${!xwallet}""${!xwork}" -p "$MINER_PWD" -i "${!xintensity}" $CCMINER_OPTS ${!xext}"
+  echo '$LAUNCH $HCD $UCCALGO -o ${!xproto}://"${!xpool}":"${!xport}" -u "${!xaddr}""${!xwallet}""${!xwork}" -p "$MINER_PWD" -i "${!xintensity}" $CCMINER_OPTS ${!xext}'
```

Comment:
There is something bad with quotes here, what you expected to print? The single quote will prevent any variable expansion and will print a string with missing opening quote for the first string
$LAUNCH $HCD $UCCALGO -o ${!xproto}://"
and missing closing quote for the last one
" $CCMINER_OPTS ${!xext}

Reply:
When miner fails to start the echo from 3main does not show miner command 
So I thought better print miner command from 0miner instead 
Doesnt multiple double quotes mix up?

Reply:
Yes, exactly, but when you use a single quote everything inside is not expanded. To obtain the same result as at line 250, for example, you need to use double quotes and avoid mixups by removing them when they occur in the middle:

echo "$LAUNCH $HCD $UCCALGO -o ${!xproto}://${!xpool}:${!xport} -u ${!xaddr}${!xwallet}${!xwork} -p $MINER_PWD -i ${!xintensity} $CCMINER_OPTS ${!xext}"

Reply:
But, to give that echo a sense of debugging I would rather keep the same quoting as in the eval, so just repeat the very same command with echo in place of eval without altering quotes. If there is
eval $LAUNCH $HCD
just do
echo $LAUNCH $HCD
and not 
echo "$LAUNCH $HCD"

Reply:
I was afraid echo $LAUNCH $HCD would launch the miner again
Doesnt it?

Reply:
nope, it's just an echo ;-)

Reply:
Same for the long ones or keep them with double quotes at beginning and end?

Reply:
Expecially those one, because if some bug occur in the quoting, as it is more complex there, we need the echo to reflect what is actually been executed.